pkg/resources/saml2_integration.go

```diff
@@ -0,0 +1,543 @@
+
+	if d.HasChange("enabled") {
+		set.WithEnabled(d.Get("enabled").(bool))
+	}
+
+	if d.HasChange("saml2_issuer") {
+		set.WithSaml2Issuer(d.Get("saml2_issuer").(string))
+	}
+
+	if d.HasChange("saml2_sso_url") {
+		set.WithSaml2SsoUrl(d.Get("saml2_sso_url").(string))
+	}
+
+	if d.HasChange("saml2_provider") {
+		valueRaw := d.Get("saml2_provider").(string)
+		value, err := sdk.ToSaml2SecurityIntegrationSaml2ProviderOption(valueRaw)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		set.WithSaml2Provider(value)
+	}
+
+	if d.HasChange("saml2_x509_cert") {
+		set.WithSaml2X509Cert(d.Get("saml2_x509_cert").(string))
+	}
+
+	if d.HasChange("saml2_sp_initiated_login_page_label") {
+		set.WithSaml2SpInitiatedLoginPageLabel(d.Get("saml2_sp_initiated_login_page_label").(string))
+	}
+
+	if d.HasChange("saml2_enable_sp_initiated") {
+		set.WithSaml2EnableSpInitiated(d.Get("saml2_enable_sp_initiated").(bool))
+	}
+
+	if d.HasChange("saml2_snowflake_x509_cert") {
+		set.WithSaml2SnowflakeX509Cert(d.Get("saml2_snowflake_x509_cert").(string))
+	}
+
+	if d.HasChange("saml2_sign_request") {
+		set.WithSaml2SignRequest(d.Get("saml2_sign_request").(bool))
+	}
+
+	if d.HasChange("saml2_requested_nameid_format") {
+		v := d.Get("saml2_requested_nameid_format").(string)
+		if len(v) > 0 {
+			value, err := sdk.ToSaml2SecurityIntegrationSaml2RequestedNameidFormatOption(v)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			set.WithSaml2RequestedNameidFormat(value)
+		} else {
+			unset.WithSaml2RequestedNameidFormat(true)
+		}
+	}
+
+	if d.HasChange("saml2_post_logout_redirect_url") {
+		v := d.Get("saml2_post_logout_redirect_url").(string)
+		if len(v) > 0 {
+			set.WithSaml2PostLogoutRedirectUrl(v)
+		} else {
+			unset.WithSaml2PostLogoutRedirectUrl(true)
+		}
+	}
+
+	if d.HasChange("saml2_force_authn") {
+		set.WithSaml2ForceAuthn(d.Get("saml2_force_authn").(bool))
+	}
+
+	if d.HasChange("saml2_snowflake_issuer_url") {
+		set.WithSaml2SnowflakeIssuerUrl(d.Get("saml2_snowflake_issuer_url").(string))
+	}
+
+	if d.HasChange("saml2_snowflake_acs_url") {
+		set.WithSaml2SnowflakeAcsUrl(d.Get("saml2_snowflake_acs_url").(string))
+	}
+
+	if d.HasChange("allowed_user_domains") {
+		v := d.Get("allowed_user_domains").([]any)
+		userDomains := make([]sdk.UserDomain, len(v))
+		for i := range v {
+			userDomains[i] = sdk.UserDomain{
+				Domain: v[i].(string),
+			}
+		}
+		set.WithAllowedUserDomains(userDomains)
+	}
+
+	if d.HasChange("allowed_email_patterns") {
+		v := d.Get("allowed_email_patterns").([]any)
+		emailPatterns := make([]sdk.EmailPattern, len(v))
+		for i := range v {
+			emailPatterns[i] = sdk.EmailPattern{
+				Pattern: v[i].(string),
+			}
+		}
+		set.WithAllowedEmailPatterns(emailPatterns)
+	}
+
+	if d.HasChange("comment") {
+		if v := d.Get("comment").(string); len(v) > 0 {
+			set.WithComment(v)
+		} else {
+			unset.WithComment(true)
+		}
+	}
+	if !reflect.DeepEqual(*set, sdk.Saml2IntegrationSetRequest{}) {
```

Comment:
here you're doing DeepEqual and in the second one, simple equality check, is there any reason for this check to differ from the other one?

Reply:
I guess that the reason are underlying lists, so equality operator cannot be used, right?

Reply:
That's correct - DeepEqual is used because set structure contains lists. For consistency we could always use `DeepEqual`, wdyt?